### PR TITLE
Don't include dev dependency licenses

### PR DESCRIPTION
**Pull request by david4096**
**Description**

Changes the operation of license checker to only include `--production` dependencies in the license file, not development dependencies. Goes down to ~250 from where it is now 1463.

**Review Instructions**

`npm run license` should generate the CSV of only production dependencies. The CI jobs should be implicitly testing this.

**Issue**

https://github.com/dockstore/dockstore/issues/5549

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

**Review by svonworl — APPROVED** [review]

**Comment by denis-yuen** [comment]
> Changes the operation of license checker to only include `--production` dependencies in the license file, not development dependencies. Goes down to ~250 from where it is now 1463.

For the ticket, I think there is a larger question. Is this removal basically confirming that we only need a list of third party dependencies that we actually include in our distributed artiffacts?  (and that the basic package.json or maven pom files are enough to capture our developer-level dependencies)

**Review by denis-yuen — COMMENTED** [review]
question

**Comment by david4096** [comment]
> > Changes the operation of license checker to only include `--production` dependencies in the license file, not development dependencies. Goes down to ~250 from where it is now 1463.
> 
> For the ticket, I think there is a larger question. Is this removal basically confirming that we only need a list of third party dependencies that we actually include in our distributed artiffacts? (and that the basic package.json or maven pom files are enough to capture our developer-level dependencies)

Responding in the original ticket

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "build.prod": "npx ng build",
     "build": "npx ng build",
     "watch": "ng build --watch --configuration development",
-    "license": "license-checker --excludePackages dockstore-ui2@$npm_package_version --csv > THIRD-PARTY-LICENSES.csv",
+    "license": "license-checker --production --excludePackages dockstore-ui2@$npm_package_version --csv > THIRD-PARTY-LICENSES.csv",
     "circle-ci-license-test-file": "license-checker --excludePackages dockstore-ui2@$npm_package_version --csv > CIRCLE-THIRD-PARTY-LICENSES.csv",
```
Comment (coverbeck) [inline package.json:21]: I think you need to add `--production` here as well.

**Review by coverbeck — COMMENTED** [review]
You need to run `npm run license` and commit the updated license file. And make the suggested change in the package.json, or the license check will fail (it checks whether the committed THIRD-PARTY-LICENSES.csv matches what's generated from the package.json).

Also, there should probably be PR(s) for our Java repos (can be a followup ticket(s)); assuming there's a way, we should be able to remove `test` scope dependencies from those third party license files.

**Review by coverbeck — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Pull request merged**